lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -24,6 +24,122 @@
   [src getCharacters:(unichar *)&result[0] range:NSMakeRange(0, size)];
   return result;
 }
+std::u16string getDefaultLocale() {
+  // Environment variable used for testing only
+  const char *testLocale = std::getenv("_HERMES_TEST_LOCALE");
+  if (testLocale) {
+    NSString *nsTestLocale = [NSString stringWithUTF8String:testLocale];
+    return nsStringToU16String(nsTestLocale);
+  }
+  NSString *defLocale = [[NSLocale currentLocale] localeIdentifier];
+  return nsStringToU16String(defLocale);
+}
+// Implementer note: This method corresponds roughly to
+// https://tc39.es/ecma402/#sec-bestavailablelocale
+llvh::Optional<std::u16string> bestAvailableLocale(
+    const std::vector<std::u16string> &availableLocales,
+    const std::u16string &locale) {
+  // 1. Let candidate be locale
+  std::u16string candidate = locale;
+
+  // 2. Repeat
+  while (true) {
+    // a. If availableLocales contains an element equal to candidate, return
+    // candidate.
+    if (llvh::find(availableLocales, candidate) != availableLocales.end()) {
+      return candidate;
+    }
+
+    // b. Let pos be the character index of the last occurrence of "-" (U+002D)
+    // within candidate.
+    size_t pos = candidate.rfind(u'-');
+
+    // ...If that character does not occur, return undefined.
+    if (pos == std::string::npos) {
+      return u"und";
```

Comment:
The advantage of using optional here is that we can clearly distinguish `undefined`.
```suggestion
      return llvh::None;
```